core/common/tarutil.cpp

```diff
@@ -38,6 +39,115 @@ namespace fc::common {
     }
   }
 
+  void createDir(const fs::path &absolute_path, const std::string &base) {}
+
+  outcome::result<int> zipTar(const std::string &input_path) {
+    if (!fs::exists(input_path)) {
+      logger->error("Zip tar: {} doesn't exists", input_path);
+      return TarErrors::kCannotZipTarArchive;
+    }
+
+    int p[2];
+
+    if (pipe(p) < 0) {
+      return TarErrors::kCannotZipTarArchive;
+    }
+
+    std::function<void()> clear = [&]() {
+      close(p[1]);
+      close(p[0]);
+    };
+    auto _ = gsl::finally([&]() { clear(); });
+
+    auto a = ffi::wrap(archive_write_new(), archive_write_free);
+    archive_write_set_format_v7tar(a.get());
+    int r = archive_write_open_fd(a.get(), p[1]);
+    if (r < ARCHIVE_OK) {
+      if (r < ARCHIVE_WARN) {
+        logger->error("Zip tar: {}", archive_error_string(a.get()));
+        return TarErrors::kCannotZipTarArchive;
+      } else {
+        logger->warn("Zip tar: {}", archive_error_string(a.get()));
+      }
+    }
+    std::function<outcome::result<void>(const fs::path &, const fs::path &)>
```

Comment:
```suggestion
    auto
```

Reply:
It is necessary because we need recursive calls